eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/location/CurrentLocationModule.java

```diff
@@ -107,22 +105,54 @@ public void shutDown()
     @Override
     public void onSearchQueryChange(String newQuery)
     {
-        if (!newQuery.isEmpty())
+        if (newQuery.isEmpty())
         {
-            mUpdateListener.onUpdate(new Clear<ListItem>(newQuery));
+            cacheCurrentLocation();
+
+            ListItem item = new Clickable<>(
+                    new MessageItem(mActivity.getString(R.string.schedjoules_location_picker_current_location)),
+                    new OnClickAction()
+                    {
+                        @Override
+                        public void onClick()
+                        {
+                            loadCurrentLocation();
```

Comment:
A few lines above a load is started (`cacheCurrentLocation()`). It's not the perfectly optimized solution, but it looked good enough now with some advantages, namely that if screen stays on and user moves, and click, it will use the latest location. But if it is the same as when the screen loaded, it will return quickly.
So there are 2 calls involed, the first for `getLastKnownLocation()` which is actually an instant synchronous call, just in our design now it is async, but still very fast. That is always executed, not cached. The second call for `GeoCoder` to get the city (address) is the slower, that is cached for a `GeoLocation`.
This way also when user clears the query, the caching call starts again.
The caching call has no callbacks, so if there is any error there, or simply user presses in the meantime, it won't cause any problem.
As mentioned in the description, there is no progress indication currently, which is only needed if user presses right away or cache call failed maybe. So not a frequent case. I would do a improvement story for that after #175.